activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -575,7 +581,11 @@ def visit_Arel_Table(o, collector)
           end
 
           if o.table_alias
-            collector << " " << quote_table_name(o.table_alias)
+            if collector.instance_values["delegate"] && collector.instance_values["delegate"].instance_values["str"].include?("INSERT")
```

Comment:
Same here, this should not be here. We should just make sure the AST being built is valid.